fullnode/fullnode_lazy.go

```diff
@@ -0,0 +1,235 @@
+package fullnode
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/types"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+
+	sdk "github.com/pokt-network/shannon-sdk"
+)
+
+// queryCodec is a package-level codec for unmarshaling account data.
+// Initialized globally to avoid expensive repeated setup across LazyFullNode instances.
+var queryCodec *codec.ProtoCodec
+
+// init initializes the codec for the fullnode module.
+func init() {
+	reg := cdctypes.NewInterfaceRegistry()
+	accounttypes.RegisterInterfaces(reg)
+	cryptocodec.RegisterInterfaces(reg)
+	queryCodec = codec.NewProtoCodec(reg)
+}
+
+// TODO_MVP(@adshmh): Rename `LazyFullNode`: this struct does not perform any caching and should be named accordingly.
+//
+// LazyFullNode: default implementation of a full node for the Shannon.
+//
+// A LazyFullNode queries the onchain data for every data item it needs to do an action (e.g. serve a relay request, etc).
+// This is done to enable supporting short block times (a few seconds), by avoiding caching
+// which can result in failures due to stale data in the cache.
+//
+// Key differences from a caching full node:
+// - Intentionally avoids caching:
+//   - Enables support for short block times (e.g. LocalNet)
+//   - Use CachingFullNode struct if caching is desired for performance
+type LazyFullNode struct {
+	// logger polylog.Logger
```

Comment:
Why is this commented out?

No need for a detail answer here, please do one of: #PUC, TODO, uncomment, etc

Reply:
The regular full node did not use any logging so it was unused and missed by the compiler. For now I've just uncommented it.